crates/starknet-server/src/api/http/endpoints/postman.rs

```diff
@@ -1,30 +1,158 @@
-use axum::Json;
-use starknet_types::felt::TransactionHash;
+use axum::{Extension, Json};
+use starknet_rs_core::types::MsgToL1;
+use starknet_types::rpc::transactions::L1HandlerTransaction;
 
 use crate::api::http::error::HttpApiError;
 use crate::api::http::models::{
-    MessageFromL2, MessageHash, MessageToL2, PostmanLoadL1MessagingContract,
+    FlushParameters, FlushedMessages, MessageHash, MessageToL1, MessageToL2, MessagingLoadAddress,
+    PostmanLoadL1MessagingContract, TxHash,
 };
-use crate::api::http::HttpApiResult;
+use crate::api::http::{HttpApiHandler, HttpApiResult};
 
 pub(crate) async fn postman_load(
-    Json(_l1_contract): Json<PostmanLoadL1MessagingContract>,
-) -> HttpApiResult<()> {
-    Err(HttpApiError::GeneralError)
+    Extension(state): Extension<HttpApiHandler>,
+    Json(data): Json<PostmanLoadL1MessagingContract>,
+) -> HttpApiResult<Json<MessagingLoadAddress>> {
+    let mut starknet = state.api.starknet.write().await;
+
+    let messaging_contract_address = starknet
+        .configure_messaging(
+            &data.network_url,
+            data.address.as_deref(),
+            data.private_key.as_deref(),
+        )
+        .await
+        .map_err(|e| HttpApiError::MessagingError { msg: e.to_string() })?;
+
+    Ok(Json(MessagingLoadAddress { messaging_contract_address }))
 }
 
-pub(crate) async fn postman_flush() -> HttpApiResult<()> {
-    Err(HttpApiError::GeneralError)
+pub(crate) async fn postman_flush(
+    Extension(state): Extension<HttpApiHandler>,
+    Json(data): Json<FlushParameters>,
+) -> HttpApiResult<Json<FlushedMessages>> {
+    // Need to handle L1 to L2 first in case that those messages
+    // will create L2 to L1 messages.
+    let mut starknet = state.api.starknet.write().await;
+
+    let dry_run = data.dry_run.unwrap_or(false);
+
+    let messages_to_l2 = if dry_run {
+        Ok(vec![])
+    } else {
+        starknet
+            .fetch_and_execute_messages_to_l2()
+            .await
+            .map_err(|e| HttpApiError::MessagingError { msg: e.to_string() })?
+            .into_iter()
+            .map(|m| m.try_into())
+            .collect::<Result<Vec<MessageToL2>, HttpApiError>>()
+    };
+
+    let from_block = if let Some(m) = &starknet.messaging { m.last_local_block } else { 0 };
+
+    let (messages_to_l1, last_local_block) = if dry_run {
```

Comment:
Under this line, there are some cryptic variable names. I'm ok with `e` for error, but `m` and `b` are not intuitive.